bentoml/cli/__init__.py

```diff
@@ -420,6 +421,44 @@ def install_completion(append, shell, path):
                 "install it with `pip install click_completion`"
             )
 
+    @bentoml_cli.command(
+        help='Retrieves BentoService artifacts into a target directory',
+        short_help="Retrieves BentoService artifacts into a target directory",
+    )
+    @conditional_argument(pip_installed_bundle_path is None, "bento", type=click.STRING)
+    @click.option(
+        '--target_dir',
+        help="Directory to put artifacts into. Defaults to pwd.",
+        default=os.getcwd(),
+    )
+    def retrieve(bento, target_dir):
+        if ':' not in bento:
+            _echo('BentoService {bento} invalid - specify name:version')
+            return
+        name, version = bento.split(':')
+
+        yatai_client = YataiClient()
+
+        track_cli('bento-retrieve')
+        get_bento_result = yatai_client.repository.get(name, version)
+        if get_bento_result.status.status_code != status_pb2.Status.OK:
+            error_code, error_message = status_pb_to_error_code_and_message(
+                get_bento_result.status
+            )
+            raise BentoMLException(
+                f'BentoService {name}:{version} not found - '
+                f'{error_code}:{error_message}'
+            )
+            return
+
+        bento_service_bundle_path = resolve_bundle_path(
+            bento, pip_installed_bundle_path
+        )
+
+        bento_path = safe_retrieve(bento_service_bundle_path, target_dir)
+
+        click.echo('Service %s artifact directory => %s' % (name, bento_path))
```

Comment:
It looks like you were trying to achieve two things here:

1. if this bento is stored locally, return the path where it is stored
2. if this bento is stored remotely, download it and copy to target_dir

Would it make more sense to just add a separate command for printing the saved path?

Currently, we've been using `jq` to do this in our documentation:
```bash
$ bentoml get IrisClassifier:latest -q | jq -r ".uri.uri"

/Users/chaoyu/bentoml/repository/IrisClassifier/20200616144911_7CE204
```
Would be nice to add a command or option for this, for example:
```bash
$ bentoml get IrisClassifier:latest --print-location

/Users/chaoyu/bentoml/repository/IrisClassifier/20200616144911_7CE204
```
The CI pipeline can use the output to determine if this is a local path or if it needs to be downloaded with `retrieve` command:
```
$ bentoml retrieve IrisClassifier:latest --target-dir /foo/bar
```

Reply:
I like this idea of adding `--print-location` - but since this PR doesn't depend on this feature (uri info is already exposed via get as you mentioned), perhaps this addition to `get` could be added in a new issue/pr? What do you think?

Reply:
Yes, that sounds great! I mentioned it just in case it is something useful for your CI pipeline use case. I will create a separate issue for that.

Reply:
awesome, ty